src/components/dialog/dialog.js

```diff
@@ -272,8 +272,8 @@ function MdDialogProvider($$interimElementProvider) {
   function advancedDialogOptions($mdDialog) {
     return {
       template: [
-        '<md-dialog aria-label="{{ dialog.ariaLabel }}">',
-          '<md-content>',
+        '<md-dialog aria-label="{{ dialog.ariaLabel }}" tabIndex="-1">',
+          '<md-content role="document" tabIndex="0">',
```

Comment:
Yes. It creates a context for the message part of a dialog, within the `alertdialog` role. Multiple sources say so:
 * http://www.w3.org/TR/wai-aria-practices/#alert
 * https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alertdialog_role